src/foam/build/FilesJsGen.js

```diff
@@ -0,0 +1,290 @@
+        'foam/core/lib',
+      ],
+    },
+    {
+      name: 'CORE_MODELS',
+      documentation: `
+        The following models must be added to files.js after boot and in this
+        specific order.
+      `,
+      value: [
+        'foam.core.stdlibScript',
+        'foam.core.eventsScript',
+        'foam.core.ContextScript',
+        'foam.core.BootScript',
+        'foam.core.FObjectScript',
+        'foam.core.ModelScript',
+        'foam.core.Property',
+        'foam.core.Simple',
+        'foam.core.AbstractMethod',
+        'foam.core.Method',
+        'foam.core.BootPhase2',
+        'foam.core.BooleanScript',
+        'foam.core.AxiomArrayScript',
+        'foam.core.EndBootScript',
+      ],
+    },
+    {
+      name: 'PHASE_1',
+      documentation: `
+        The following models must be added to files.js second but their order
+        does not matter and all of their dependencies will be automatically
+        added first.
+      `,
+      value: [
+        'foam.core.ContextMultipleInheritenceScript',
+        'foam.core.DebugDescribeScript',
+        'foam.core.ImplementsModelRefine',
+        'foam.core.ImportExportModelRefine',
+        'foam.core.ListenerModelRefine',
+        'foam.core.MethodArgumentRefine',
+        'foam.core.ModelConstantRefine',
+        'foam.core.ModelRefinestopics',
+        'foam.core.ModelRequiresRefines',
+        'foam.core.ModelActionRefine',
+        'foam.core.Promised',
+        'foam.core.__Class__',
+        'foam.core.__Property__',
+      ],
+    },
+    {
+      name: 'PHASE_2',
+      documentation: `
+        The following models must be added to files.js after PHASE_1. Their
+        order doesn't matter as long as they're added after PHASE_1. Their
+        dependencies will automatically be added first.
+      `,
+      value: [
+        'foam.core.ModelRefinescss',
+        'foam.core.WindowScript',
+        'foam.net.WebLibScript',
+      ],
+    },
+    {
+      name: 'NANOS_MODELS',
+      documentation: `
+        These are the models needed for booting nanos.
+      `,
+      value: [
+        'foam.blob.AbstractBlobService',
+        'foam.blob.RestBlobService',
+        'foam.box.PromisedBox',
+        'foam.box.RPCReturnBox',
+        'foam.box.SocketConnectBox',
+        'foam.dao.JDAO',
+        'foam.dao.PromisedDAO',
+        'foam.nanos.notification.Notification',
+        'foam.u2.search.TextSearchView',
+      ],
+    },
+  ],
+  properties: [
+    {
+      name: 'required',
+      documentation: `
+        These are the models to be loaded when files.js is finished loading.
+      `,
+      factory: function() { return this.NANOS_MODELS },
+    },
+    {
+      name: 'srcDir',
+      value: 'STRIPPED/src',
+    },
+    {
+      name: 'modelDAO',
+      expression: function(srcDir) {
+        return this.DirCrawlModelDAO.create({srcDir: srcDir})
+      },
+    },
+  ],
+  methods: [
+    function execute() {
+      this.getFilesJs().then(console.log);
```

Comment:
I think you should console.log.bind(console), its not pre-bound on all systems.

Reply:
Done